src/replication.c

```diff
@@ -1897,23 +1921,25 @@ empty_mgmt_q_of_replica(replica_t *r)
 	mgmt_cmd_t *mgmt_cmd;
 	while ((mgmt_cmd = TAILQ_FIRST(&r->mgmt_cmd_queue))) {
 		mgmt_cmd->io_hdr->status = ZVOL_OP_STATUS_FAILED;
-		switch (mgmt_cmd->io_hdr->opcode) {
-			case ZVOL_OPCODE_SNAP_CREATE:
-				handle_snap_create_resp(r, mgmt_cmd);
-				break;
-			case ZVOL_OPCODE_PREPARE_FOR_REBUILD:
-				handle_prepare_for_rebuild_resp(r->spec,
-				    mgmt_cmd->io_hdr, NULL, mgmt_cmd);
-				break;
-			case ZVOL_OPCODE_STATS:
-				update_spec_stats(r->spec, mgmt_cmd->io_hdr,
-				    NULL);
-				break;
-			default:
-				break;
+		if (!mgmt_cmd->cmd_completed) {
+			switch (mgmt_cmd->io_hdr->opcode) {
+				case ZVOL_OPCODE_SNAP_CREATE:
+					handle_snap_create_resp(r, mgmt_cmd);
+					break;
+				case ZVOL_OPCODE_PREPARE_FOR_REBUILD:
+					handle_prepare_for_rebuild_resp(r->spec,
+					    mgmt_cmd->io_hdr, NULL, mgmt_cmd);
+					break;
+				case ZVOL_OPCODE_STATS:
+					update_spec_stats(r->spec, mgmt_cmd->io_hdr,
+					    NULL);
+					break;
+				default:
+					break;
+			}
+			REPLICA_NOTICELOG("mgmt command(%d) failed for replica(%lu)\n",
+			    mgmt_cmd->io_hdr->opcode, r->zvol_guid);
 		}
-		REPLICA_NOTICELOG("mgmt command(%d) failed for replica(%lu)\n",
```

Comment:
lets add a LOG here

Reply:
Here we have two types of management command...
1. Commands which were queued in mgmt queue and haven't received any response.
2. A command for which we have already received a response but due to multiple responses `read_io_resp` has reported an error and didn't dequeue it from mgmt command queue.

For 1st type, we are logging error in `empty_mgmt_q_of_replica` and for 2nd type, we are logging it as `protocol error` in `read_io_resp`.